fix: top Table items were invisible after window resize

cmitsakis opened the pull request:
<!-- If this is your first pull request for Fyne please read the contributor docs at:
https://github.com/fyne-io/fyne/wiki/Contributing.
Be sure that your work is based off `develop` branch. --> 

### Description:
Fixes #3034 

This appears to be working.

Unfortunately I couldn't write tests for this one because I don't know how to check for visible elements, but I decided to send the PR now hoping it will be merged before the next release.

The issue was happening only when the size of the table increased as a result of the window size increasing. So in that case only I call ScrollToTop().

Losing scroll position doesn't seem to be a problem. The top row appears anyway, by calling ScrollToTop() I make it visible. But there might be cases where it causes a small inconvenience.

### Checklist:
<!-- Please tick these as appropriate using [x] -->

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

Jacalz reviewed (changes requested):
Thanks for the PR. It would be good to add a test to make sure that the issue doesn’t appear again.

cmitsakis commented:
In Table it's not possible to access the cells because they are stored in TableCellsRenderer. That's why I couldn't create a test (although I created a test for my fix in List where I could access the items).

But I found a workaround that works. I store every label created by CreateCell() in a slice and check the text of those labels. It seems to work although I don't know if it's a good idea and I don't really understand it.

andydotxyz commented on widget/table.go:
```diff
@@ -512,7 +512,13 @@ func (c *tableCells) Resize(s fyne.Size) {
+	if c.size.Height > cSizeBeforeResize.Height {
+		// table height has increased.
+		// call ScrollToTop() otherwise empty rows will appear at the top of the table.
+		c.t.ScrollToTop()
```
I see that this fixes some instances of the issue, but if there are 1000 items and you are looking at 900 and the view gets larger then scrolling to the top won't be the right thing to do

cmitsakis replied:
I noticed that the if condition becomes true only when the window is resized to a size big enough that there are no hidden items. So `ScrollToTop()` doesn't cause any problems.
But there should be a better fix. In my comment below I describe what I noticed about the value of `c.size.Height`.

andydotxyz replied:
Your explanation makes sense - but it does not match what the code is doing now, so it will still scroll up some times when it is not desired.
I agree there is likely a better fix, have you any ideas?

cmitsakis replied:
I agree there is likely a better fix, have you any ideas?
here is an idea I tried at first: https://github.com/fyne-io/fyne/commit/cfaeb96b6476d2d254809e19858f4e62cb736b94
I deleted it by force pushing because I noticed it doesn't work when I make the window smaller. but if I can fix it further it might be a better solution.

andydotxyz commented on widget/table_test.go:
```diff
@@ -572,3 +572,58 @@ func (t *separatorThicknessZeroTheme) Size(n fyne.ThemeSizeName) float32 {
+	w.Resize(fyne.NewSize(100, 50))
+	table.ScrollToBottom()
+	table.Select(TableCellID{Row: maxRows - 1, Col: 1})
+	// increase size big enough that there is no scroll
```
Can we trigger it with a smaller increase, so it handles the case where there should still be hidden items, but that some visible ones are not showing up?

cmitsakis replied:
I believe it's not possible to trigger this bug if there are hidden items.

andydotxyz reviewed (changes requested):
Thanks for looking at this, but as you say there probably needs to be a more accurate fix for this issue

cmitsakis commented:
I improved the test. I found out how to calculate the exact size that triggers this bug.  Although when I run the test with -race it doesn't work as expected. very strange.

Also when I added this line
```go
log.Printf("c.size.Height (%v) > cSizeBeforeResize.Height (%v)\n", c.size.Height, cSizeBeforeResize.Height)
```
in Resize() before I call `c.t.ScrollToTop()`, I noticed that before I increase the size of the window c.size.Height = 109 and after I increase it c.size.Height = 110

So I'm wondering what would happen if I fixed the size to be 110 instead of 109. Not sure where this value is calculated.

I will look into it again tomorrow to see if I can improve my fix.

Jacalz commented:
Please don’t force-push your changes. It messes with the commit history and make everything harder to review. We’ll squish when merging if we think it’s needed.

andydotxyz commented:
This fix was covered in a separate PR at some point, closing this change.
